packages/aws-cdk-lib/aws-cloudwatch/lib/variable.ts

```diff
@@ -0,0 +1,171 @@
+export enum VariableInputType {
```

Comment:
Assuming this and the `enum` below are what you're talking about @humanzz. For the decision between enums and enum-like classes, I consider two things:

- Is it possible/likely for additional values to be added later?
- If an additional value was to be added, would it involve changes elsewhere in the CDK?

 I can't think of an additional `VariableInputType` that would come in the future since these three UI options seem pretty standard to me. 

For `VariableType`, if a new option came in, it likely would involve changing additional things around the CDK to support whatever new type it is, since that feels like it would be a pretty significant change. Basically, it doesn't feel like it would be as simple as adding in a new `VariableType` and calling it a day. It feels like the new `VariableType` would be a new use case unlocked that would have ramifications on other properties in CDK land as well.

If any of these assumptions are off base let me know! I'm definitely not married to `enums`, but that's why I didn't make a fuss.

Reply:
Thanks for this explanation. It makes sense to me :)